### PR TITLE
Library setup

### PR DESCRIPTION
**Pull request by UsirusenIkon**
In this PR, we did the following:

### Create class Person with the following:
- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and - parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

### Create class Student with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

### Create class Teacher with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def can_use_services?
+    of_age || parent_permision
+  end
+
+  private
+
+  def of_age?
+    of_age >= 18
+  end
```
Comment (ichala) [inline person.rb]: - [ ]  kindly update your can_use_services method to use the instance variable @parent 

```Ruby
  def can_use_services?
    of_age || @parent
  end
```

**Review by ichala — CHANGES_REQUESTED** [review]
Hi @UsirusenIkon 
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No linter errors ✔️ 
- Correct Workflow ✔️ 


 ## Required Changes ♻️


- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    @age = age
+    @parent = parent_permision
+  end
+
+  attr_accessor :@name, :@age
+
+  att_reader :@id
+
+  def can_use_services?
+    of_age || @parent_permision
```
Comment (mckent05) [inline person.rb]: - Like the previous reviewer said please use the variable you assigned to the `parent_permission` which is `@parent`. Please fix this 🙏 
- **OPTIONAL** - You named the method to check the age as `of_age?` but you are calling a `of_age` method. Please fix this 🙏 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @UsirusenIkon, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    @age = age
+    @parent = parent_permision
+  end
+
+  attr_accessor :@name, :@age
+
+  att_reader :@id
+
+  def can_use_services?
+    of_age || @parent
```
Comment (TedLivist) [inline person.rb:13]: - [ ] **[OPTIONAL]** I would recommend you use `of_age?` (without question mark) as that is what you have.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @UsirusenIkon,

✔️ Implementation of inheritance

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**